### DISCUSSION TITLE
About the number of workspace (there seems to be a mistake) && Installation failed without any output

### DISCUSSION THREAD

**Discussion by arleyGuoLei** [pnpm/pnpm#4273] [Q&A]
### something background ↓

- pnpm version: `6.25.1`

- pnpm-workspace.yaml

```yaml
packages:
  - 'packages/**'
  - '!**/example/**'
```

- my project dirs

```text
├── example
|  └── react-koa
├── node_modules
├── package.json
├── packages
|  ├── cli
|  ├── core-koa
|  ├── core-react
|  ├── plugin-koa
|  ├── plugin-react
|  ├── types
|  ├── utils
|  └── webpack
├── pnpm-lock.yaml
├── pnpm-workspace.yaml
```

### 1. Installation failed without any output

`example/react-koa` is not a workspace, I did the installation in the directory but didn't download node_modules ... output ↓

![image](https://user-images.githubusercontent.com/40717741/150497486-3f04e4e9-f9ba-425f-bd37-4c046fe51174.png)

And then there were no files of node_modules and `pnpm-lock.yaml`, **why**?

**I want to use pnpm to manage dependencies and be able to switch local packages during development, or directly test online packages, similar to automatic link or unlink** Any recommendations ?

### 2: About the number of workspace 

- pnpm version: `6.25.1`

- pnpm-workspace.yaml

```yaml
packages:
  - 'packages/**'
  - '!**/example/**'
```

```text
.... packages # my packages workspace
├── cli
├── core-koa
├── core-react
├── plugin-koa
├── plugin-react
├── types
├── utils
└── webpack
```

There are 8 folders in total, but the number of outputs is nine ↓ why? **Is there a way to print all workspaces name**?

![image](https://user-images.githubusercontent.com/40717741/150495536-de7b70b5-ed3f-44a9-94ce-c1cb0272798f.png)

**Comment by zkochan** [comment]
> There are 8 folders in total, but the number of outputs is nine ↓ why? Is there a way to print all workspaces name?

The root `package.json` also counts as a workspace.

And to print the list, you  may run 
```
pnpm list -r --depth=-1
```
